lang/python/python/patches/015-openssl-deprecated.patch

```diff
@@ -0,0 +1,67 @@
+--- a/Modules/_ssl.c
++++ b/Modules/_ssl.c
+@@ -94,13 +94,13 @@ struct py_ssl_library_code {
+ #include "_ssl_data.h"
+ 
+ #if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && !defined(LIBRESSL_VERSION_NUMBER)
+-#  define OPENSSL_VERSION_1_1 1
+-#  define PY_OPENSSL_1_1_API 1
++# define OPENSSL_VERSION_1_1 1
++# define PY_OPENSSL_1_1_API 1
+ #endif
+ 
+ /* LibreSSL 2.7.0 provides necessary OpenSSL 1.1.0 APIs */
+ #if defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER >= 0x2070000fL
+-#  define PY_OPENSSL_1_1_API 1
++# define PY_OPENSSL_1_1_API 1
+ #endif
+ 
+ /* Openssl comes with TLSv1.1 and TLSv1.2 between 1.0.0h and 1.0.1
+@@ -162,6 +162,11 @@ struct py_ssl_library_code {
+ 
+ #define ASN1_STRING_get0_data ASN1_STRING_data
+ #define TLS_method SSLv23_method
++#define X509_getm_notBefore X509_get_notBefore
++#define X509_getm_notAfter X509_get_notAfter
++#define OpenSSL_version_num SSLeay
++#define OpenSSL_version SSLeay_version
++#define OPENSSL_VERSION SSLEAY_VERSION
```

Comment:
I didn't take a really close look, but it looks to me that if you remove these defines and the patches below, the code is virtually unchanged

Reply:
Those defines are for OpenSSL 1.0.2. They are guarded by an if statement.

Reply:
ok;
i found similar patches like this, but they were implemented slightly different